fip-0026.md

```diff
@@ -126,199 +127,267 @@ Holds the public key of the account that will be used to hold all the domains wh
-|marketplace|Yes|name of marketplace|Name of marketplace that will receive listing fee|
-|notif_address|No|Address|Optional address, if supplied will be used to send a notification via a FIO request when the domain listing is almost expired and when the listing is cancelled|
+|max_fee|Yes|Positive Int|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by [/get_fee](https://developers.fioprotocol.io/api/api-spec/reference/get-fee/get-fee) for correct value.|
+|tpid|Yes|FIO Address|FIO Address of the entity which generates this transaction. TPID rewards will be paid to this address. Set to empty if not known.|
 
 ##### Example
 ```
 {
-	"actor": "2odzomo2v4pe"
-	"fio_domain": "alice",
-	"sale_price": 20000000000,
-	"marketplace": "marketplace",
-	"notif_address": "forsale@myotherdomain"
+    "actor": "2odzomo2v4pe"
+    "fio_domain": "alice",
+    "sale_price": 20000000000,
+    "max_fee": 1000000000,
+    "tpid": "rewards@wallet"
 }
 ```
 #### Processing
 - Request is validated per exception handling
 - Hash both the `actor`, `fio_domain`, `marketplace` to store along side the string, for indexing.
+- Retrieve current `commission_fee` from the marketplace config table to be saved with the listing
 - Add domain to the `domainsale` table with relevant information
 - Transfer domain from seller to the holder account public key.
 - A listing fee is collected by the marketplace and is set in the marketplace config table
+- Verify tx does not exceed max transaction size.
 
 #### Exception handling
 |Error condition|Trigger|Type|fields:name|fields:value|Error message|
 |---|---|---|---|---|---|
-|Invalid marketplace|Supplied `marketplace` is not found in marketplace config table `mrkplconfig`|400| |"Marketplace not found"|
-|Invalid FIO domain|Supplied `fio_domain` is not registered.|400|"fio_domain"| |"FIO domain not found"|
-|Invalid sale price|Supplied `sale_price` (in SUF) is not greater than 1|400|"sale_price"| |"Sale price must be greater than 1"|
-|FIO Domain not owned by actor|Supplied `fio_domain` is owned by the actor.|400|"actor"| |"FIO domain not owned by actor"|
+|Invalid marketplace|Supplied `marketplace` is not found in marketplace config table `mrkplconfig`|400|"marketplace"|Value sent in, e.g. "notamarketplace"|"Marketplace not found"|
+|Invalid sale price|Supplied `sale_price` (in SUF) is not greater than 1|400|"sale_price"|Value sent in, e.g. "1000000000"|"Sale price must be greater than 1"|
+|Invalid fee value|max_fee format is not valid|400|"max_fee"|Value sent in, e.g. "-100"|"Invalid fee value"|
+|Fee exceeds maximum|Actual fee is greater than supplied max_fee|400|"max_fee"|Value sent in, e.g. "1000000000"|"Fee exceeds supplied maximum"|
+|Invalid TPID|tpid format is not valid|400|"tpid"|Value sent in, e.g. "notvalidfioaddress"|"TPID must be empty or valid FIO address"|
+|Invalid FIO domain|Supplied `fio_domain` is not listed for sale.|400|"fio_domain"|Value sent in, e.g. "alice"|"FIO domain not found"|
```

Comment:
I didn't plan on searching the table to see if a given domain is already listed. The user simply cannot list a domain he doesn't own and by listing a domain he is no longer the owner of it because it is transferred to the holder account.

Reply:
When will "Supplied fio_domain is not listed for sale." be triggered?